### PR TITLE
Add NodeBalancer-related data sources

### PR DESCRIPTION
**Pull request by LBGarber**
This pull request adds the following data sources:
- `linode_nodebalancer`
- `linode_nodebalancer_config`
- `linode_nodebalancer_node`

### CODE REVIEW

File: linode/data_source_linode_nodebalancer_config.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			},
+			"ssl_fingerprint": {
+				Type:        schema.TypeString,
+				Description: "The read-only fingerprint automatically derived from the SSL certificate assigned to this NodeBalancerConfig. Please refer to this field to verify that the appropriate certificate is assigned to your NodeBalancerConfig.",
+				Computed:    true,
+			},
+			"node_status": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
```
Comment (Charliekenney23) [inline linode/data_source_linode_nodebalancer_config.go:103]: This resource schema could also be reused

File: linode/data_source_linode_nodebalancer_config_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+					resource.TestCheckNoResourceAttr(resName, "ssl_key"),
+				),
+			},
+		},
+	})
+}
+
+func testDataSourceLinodeNodeBalancerConfigBasic(nodeBalancerName string) string {
+	return testAccCheckLinodeNodeBalancerConfigBasic(nodeBalancerName) + `
+data "linode_nodebalancer_config" "foofig" {
```
Comment (Charliekenney23) [inline linode/data_source_linode_nodebalancer_config_test.go:51]: I love `foofig` lol

File: linode/data_source_linode_nodebalancer_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLinodeNodeBalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceLinodeNodeBalancerBasic(nodebalancerName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLinodeNodeBalancerExists,
```
Comment (Charliekenney23) [inline linode/data_source_linode_nodebalancer_test.go:23]: make sure transfer is set here

**Review by Charliekenney23 — CHANGES_REQUESTED** [review]

File: linode/data_source_linode_nodebalancer_config_test.go (hunk 2 above)
Comment (LBGarber) [reply linode/data_source_linode_nodebalancer_config_test.go:51]: Same lol

**Review by Charliekenney23 — APPROVED** [review]

**Pull request merged**